scarb/src/compiler/plugin/proc_macro/host.rs

```diff
@@ -0,0 +1,64 @@
+use crate::compiler::plugin::proc_macro::ProcMacroInstance;
+use crate::core::Package;
+use anyhow::Result;
+use cairo_lang_defs::plugin::{MacroPlugin, MacroPluginMetadata, PluginResult};
+use cairo_lang_semantic::plugin::PluginSuite;
+use cairo_lang_syntax::node::ast::ModuleItem;
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use smol_str::SmolStr;
+use std::collections::HashMap;
+use std::sync::Arc;
+use typed_builder::TypedBuilder;
+
+/// A Cairo compiler plugin controlling the procedural macro execution.
+///
+/// This plugin decides which macro plugins (if any) should be applied to the processed AST item.
+/// It then redirects the item to the appropriate macro plugin for code expansion.
+#[derive(Debug, TypedBuilder)]
+pub struct ProcMacroHost {
+    macros: HashMap<SmolStr, Arc<ProcMacroInstance>>,
+}
```

Comment:
Are you going to add a lot of fields later on to this structure? Because otherwise I suggest doing the classical `new` constructor method. If you can make stuff simpler then why not? TypedBuilder also will probably slow down compilation time and code size.